sound/soc/intel/boards/sof_cs42l42.c

```diff
@@ -340,11 +345,15 @@ static int create_spk_amp_dai_links(struct device *dev,
 	links[*id].cpus = &cpus[*id];
 	links[*id].num_cpus = 1;
 
-	links[*id].cpus->dai_name = devm_kasprintf(dev, GFP_KERNEL,
-						   "SSP%d Pin", ssp_amp);
-	if (!links[*id].cpus->dai_name) {
-		ret = -ENOMEM;
-		goto devm_err;
+	if (sof_cs42l42_quirk & SOF_CS42L42_VIRTUAL_SSP_DAI)
+		links[*id].cpus->dai_name = "SSP-SPK Pin";
```

Comment:
that's a problem here, if the changes modifies both the dailink name and the cpu dai name. we will not be able to power manage the SSP in future platforms.

Reply:
> I really don't think this can work if you also change the cpu dai name @brentlu
> 
> Changing the dailink name is one thing, it has no bearing on the rest. The change of the cpu dai name is problematic.

It's ok not to touch the dai name because what makes the 'SSPX Pin' represents a specific SSP port is the topology, not the dai name string.

Reply:
that's only true for existing products, this is not true moving forward (the host will have to power-manage each SSP), and in addition there's still the issue of backwards-compatibility. This isn't going to work @brentlu